build.rs

```diff
@@ -232,9 +232,10 @@ fn build_lib(compiled_libraries: &mut HashSet<Libs>, target: &String, lib: Libs)
         .flag_if_supported("/wd9002");
 
     if target.contains("x86_64") || target.contains("i686") {
-        build
-            .flag_if_supported("-msse2") // GNU
-            .flag_if_supported("-arch:SSE2"); // MSVC
+        build.flag_if_supported("-msse2"); // GNU
+        if target.contains("windows") {
+            build.flag_if_supported("-arch:SSE2"); // MSVC
+        }
```

Comment:
So this should probably be based on the compiler rather than the operating system, I think. You can compile this code with clang on Windows as well.